File: Modules/SMES/Esop/SIE.Web.ESop/Documents/Commands/DocumentSaveCommand.cs
using SIE.Common.Attachments;
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.ESop.Documents;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.ESop.Documents.Commands
{
    /// <summary>
    /// 保存
    /// </summary>
    [JsCommand("SIE.Web.ESop.Documents.Commands.DocumentSaveCommand")]
    public class DocumentSaveCommand : FormSaveCommand
    {
        /// <summary>
        /// 保存操作
        /// </summary>
        /// <param name="entity">发放单实体</param>
        protected override void DoSave(Entity entity)
        {
            var documentCollection = entity as DocumentCollection;

            if (documentCollection == null)
            {
                throw new ValidationException("保存数据异常，请重新尝试！".L10N());
            }
            RF.Save(documentCollection);
        }
    }
}

File: Modules/SMES/Esop/SIE.Web.ESop/Documents/Commands/DocumentSaveCommand.js
SIE.defineCommand('SIE.Web.ESop.Documents.Commands.DocumentSaveCommand',
    {
        extend: 'SIE.cmd.Save',
        meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
        canExecute: function (view) {
            var result = view.getData().isDirty() || view.getData().dirty;
            return result;
        },
        execute: function (view, source) {

            var me = this;
            me.doSave(view);
        },
        onSaved: function (view, res) {
            var me = this;
            me.onSavedMsg();
            me.view.getCurrent().markSaved();
        },
        onSavedMsg: function (view, res) {

            Ext.Msg.show({
                title: '提示'.t(),
                message: '保存成功'.t(),
                buttons: Ext.MessageBox.OK,
                icon: Ext.Msg.INFO,
                callback: function () {
                    //CRT.Workbench.closeCurrentTab();
                    CRT.Event.fire("SIE.ESop.Documents.DocumentCollection_refresh");
                }
            });
        }
    });